usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh

```diff
@@ -26,7 +26,23 @@ extract_partitions() {
     declare sysfs_name=$(get_sysfs_name $device)
 
     ### check if we can find any partitions
-    declare -a sysfs_paths=(/sys/block/$sysfs_name/$sysfs_name*)
+    declare -a sysfs_paths_unfiltered=(/sys/block/$sysfs_name/$sysfs_name*)
+
+    ### initialize array
+    declare -a sysfs_paths=()
+
+    ### Silently skip invalid partitions on eMMC devices
+    ### of type *rpmb or *boot[0-9] like /dev/mmcblk0rpmb or /dev/mmcblk0boot0
+    ### because sysfs for (some?) eMMC disks recognises partitions,
+    ### which are no usual partitions, but special areas on the eMMC.
+    ### When trying to get the partition layout for such disks, ReaR would exit with an error.
+    ### Therefore we ignore these special partitions, see also
+    ### https://github.com/rear/rear/issues/2087
+    for possible_sysfs_partition in ${sysfs_paths_unfiltered[@]}; do
+	if [[ ! ( $possible_sysfs_partition = *'/mmcblk'[0-9]'rpmb' || $possible_sysfs_partition = *'/mmcblk'[0-9]'boot'[0-9] ) ]] ; then
```

Comment:
The if condition would accept such partitions when they contain more than one number like
`/dev/mmcblk12rpmb` or `/dev/mmcblk12boot0` or `/dev/mmcblk0boot12`
but I don't know if that could ever happen in practice?

Reply:
True. It would also accept such partitions on devices other than eMMC (I do not know if such devices exist). Having more than 10 eMMCs in a system is rather unlikely I guess, but theoretically possible.

Reply:
@fabz5 
in ReaR the extglob and nullglob shell options
are enabled by the main script usr/sbin/rear
```
shopt -s nullglob extglob
```

Reply:
You are thinking of something like

```
*'/mmcblk'+([0-9])'boot'+([0-9])
```
?

Reply:
How unfortunate:
It seems that with byash 3.x on my SLES11-SP4 system
extended pattern matching as in ``
does not work
```
# ( shopt -s nullglob extglob ; possible_sysfs_partition=/dev/mmcblk12boot34 ; if [[ ! ( $possible_sysfs_partition = *'/mmcblk'+([0-9])'rpmb' || $possible_sysfs_partition = *'/mmcblk'+([0-9])'boot'+([0-9])* ) ]] ; then echo $possible_sysfs_partition is possible ; else echo $possible_sysfs_partition is not possible ; fi )
-bash: unexpected token `(', expected `)'
-bash: syntax error near `*'/mmcblk'+(['
```
while the same works with bash 4.x on my openSUSE Lap 15.o system
```
# ( shopt -s nullglob extglob ; possible_sysfs_partition=/dev/mmcblk12boot34 ; if [[ ! ( $possible_sysfs_partition = *'/mmcblk'+([0-9])'rpmb' || $possible_sysfs_partition = *'/mmcblk'+([0-9])'boot'+([0-9])* ) ]] ; then echo $possible_sysfs_partition is possible ; else echo $possible_sysfs_partition is not possible ; fi )
/dev/mmcblk12boot34 is not possible
```

Reply:
Just to make sure: Could this be the problem? At least for bash 3.x?
"extglob changes the way certain characters are parsed. It is necessary to have a newline (not just a semicolon) between shopt -s extglob and any subsequent commands to use it."

Source: https://mywiki.wooledge.org/glob

Reply:
Un-be-liev-able!
You are right:
With separated commands it even works with bash 3.x on my SLES11-SP4 system:
```
# shopt -s nullglob extglob

# possible_sysfs_partition=/dev/mmcblk12boot34

# if [[ $possible_sysfs_partition = *'/mmcblk'+([0-9])'boot'+([0-9])* ]] ; then echo $possible_sysfs_partition is not possible ; else echo $possible_sysfs_partition is possible ; fi
/dev/mmcblk12boot34 is not possible

# if [[ ! ( $possible_sysfs_partition = *'/mmcblk'+([0-9])'rpmb' || $possible_sysfs_partition = *'/mmcblk'+([0-9])'boot'+([0-9])* ) ]] ; then echo $possible_sysfs_partition is possible ; else echo $possible_sysfs_partition is not possible ; fi
/dev/mmcblk12boot34 is not possible
```

Reply:
Great. Then I will commit a fix later on.

Reply:
Done

Reply:
Thank you!